data-reporting-service/investigation-service/src/main/java/gov/cdc/etldatapipeline/investigation/util/ProcessInvestigationDataUtil.java

```diff
@@ -0,0 +1,203 @@
+                for(JsonNode node : observationNotificationIdsJsonArray) {
+                    String sourceClassCode = node.get("source_class_cd").asText();
+                    String actTypeCode = node.get("act_type_cd").asText();
+
+                    if(sourceClassCode.equals("OBS") && actTypeCode.equals("PHCInvForm")) {
+                        investigationTransformed.setPhcInvFormId(node.get("source_act_uid").asLong());
+                    }
+                    if(sourceClassCode.equals("NOTF") && actTypeCode.equals("Notification")) {
+                        investigationNotification.setInvestigationId(node.get("public_health_case_uid").asLong());
+                        notificationIds.add(node.get("source_act_uid").asLong());
+                    }
+                    if(sourceClassCode.equals("OBS") && actTypeCode.equals("LabReport")) {
+                        investigationObservation.setInvestigationId(node.get("public_health_case_uid").asLong());
+                        observationIds.add(node.get("source_act_uid").asLong());
+                    }
+                }
+                for(Long id : notificationIds) {
+                    investigationNotification.setNotificationId(id);
+                    kafkaTemplate.send(investigationNotificationOutputTopicName, investigationObservation.toString());
+                }
+                for(Long id : observationIds) {
+                    investigationObservation.setObservationId(id);
+                    kafkaTemplate.send(investigationObservationOutputTopicName, investigationObservation.toString());
+                }
+            }
+        } catch (Exception e) {
+            logger.error("Error processing Observation Notification Ids JSON array from investigation data: {}", e.getMessage());
+        }
+    }
+
+    private void transformInvestigationConfirmationMethod(String investigationConfirmationMethod, InvestigationTransformed investigationTransformed, ObjectMapper objectMapper) {
+        try {
+            JsonNode investigationConfirmationMethodJsonArray = investigationConfirmationMethod != null ? objectMapper.readTree(investigationConfirmationMethod) : null;
+            InvestigationConfirmationMethod investigationConfirmation = new InvestigationConfirmationMethod();
+            Long investigationId = null;
+            Map<String, String> confirmationMethodMap = new HashMap<>();
+            Instant confirmationMethodTime = null;
+            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+
+            // Redundant time variable in case if confirmation_method_time is null in all rows of the array
+            String phcLastChgTime = null;
+
+
+            if(investigationConfirmationMethodJsonArray != null && investigationConfirmationMethodJsonArray.isArray()) {
+                investigationId = investigationConfirmationMethodJsonArray.get(0).get("public_health_case_uid").asLong();
```

Comment:
Consider a null check

Reply:
This investigation id is not used anywhere but to set it in the InvestigationTransformed object. Even if it is coming in as null, that is okay because the default value is null!